iree/turbine/kernel/wave/wave.py

```diff
@@ -175,6 +174,16 @@ def __init__(
 
         self.grid_type = Grid[tuple(get_grid_shape(self.workgroup_constraints))]
 
+        # TODO: needed for the wave_runtime grid calculations, we should really
+        # just generate host wrapper suitable for wave_runtime instead of doing
+        # it in python (and it will be faster as well).
+        hints = get_type_hints(eager_function)
```

Comment:
Wait why do we need type hints? Seems brittle to depend on type hints to get name and arg?

Reply:
See my prev comment about `MAX_EXTEND_SEQ_LEN: tkl.SymbolBind[tkl.i32]`.

Reply:
Makes sense. Are there any performance slowdowns caused by this function vs using __annotations__ ?

Reply:
Nothing I immediately noticed. Also, this code is only called once when we create `LaunchableWave` object from `@tkw.wave` decorator, not when we actually invoking kernel, so it isn't on the hot path anyway.